pkg/progress/progress_watcher.go

```diff
@@ -0,0 +1,124 @@
+package progress
+
+import (
+	"bufio"
+	"fmt"
+	"log"
+	"os/exec"
+	"strconv"
+	"strings"
+	"time"
+)
+
+type Watcher struct {
+	progressPercents int
+	exitCode         int
+	buf              *bufio.Reader
+	subscribers      []func(ProgressState)
+	done             chan bool
+	cmd              *exec.Cmd
+}
+
+func CmdWithProgressWatching(name string, arg ...string) (*Watcher, error) {
```

Comment:
A few lines of documentation won't hurt anyone :-D

```suggestion
// CmdWithProgressWatching starts cmd and returns watchers that allows to track the progress
func CmdWithProgressWatching(name string, arg ...string) (*Watcher, error) {
```

Reply:
Applied